content/blog/lease-read.md

```diff
@@ -0,0 +1,37 @@
+---
+title: How TiKV Uses "Lease Read" to Guarantee High Performances, Strong Consistency and Linearizability
+date: 2020-02-27
+author: Siddon Tang
+---
+
+TiKV, an open source distributed transactional key-value store (also a Cloud Native Computing Foundation incubating project), uses the [Raft](https://raft.github.io/) consensus algorithm to ensure strong data consistency and high availability. Raft bears many similarities to other consensus algorithms like Paxos in its ability to ensure fault-tolerance and its performance implementations, but generally easier to understand and implement. While our team has written extensively on how Raft is used in TiKV (some examples: [Raft in TiKV](https://pingcap.com/blog/2017-07-28-raftintikv/), [the Design and Implementation of Multi-raft](https://pingcap.com/blog/2017-08-15-multi-raft/), [Raft Optimization](https://pingcap.com/blog/optimizing-raft-in-tikv/)), one topic we haven’t discussed is when stale read happens due to a brain split within a Raft group, and the old Raft leader is not aware that a new leader is elected, what should we do? This post discusses three different approaches to this problem: Raft Log Read, `ReadIndex` Read, and Lease Read, and why TiKV adopts the Lease Read approach.
```

Comment:
add a link to this  https://www.cncf.io/blog/2019/05/21/toc-votes-to-move-tikv-into-cncf-incubator/